core/templates/pages/contributor-dashboard-page/services/contribution-and-review.service.spec.ts

```diff
@@ -13,96 +13,385 @@
+
+    beforeEach(() => {
+      onSuccess = jasmine.createSpy(
+        'onSuccess', (suggestionId: string) => {});
+      onFailure = jasmine.createSpy('onFailure', () => {});
+    });
+
+    it('should call onSuccess function on' +
+    'resolving suggestion to skill correctly', fakeAsync(() => {
+      spyOn(
+        carbas, 'reviewSkillSuggestionAsync'
+      ).and.returnValue(Promise.resolve());
+
+      cars.resolveSuggestiontoSkill(
+        'abc', 'pqr', 'accept', 'review message', 'easy', onSuccess, onFailure);
+      tick();
+
+      expect(carbas.reviewSkillSuggestionAsync)
+        .toHaveBeenCalledWith('abc', 'pqr', requestBody);
+      expect(onSuccess).toHaveBeenCalledWith('pqr');
+      expect(onFailure).not.toHaveBeenCalled();
+    }));
+
+    it('should call onFailure function when' +
+    'resolving suggestion to skill fails', fakeAsync(() => {
+      spyOn(
+        carbas, 'reviewSkillSuggestionAsync'
+      ).and.returnValue(Promise.reject());
+
+      cars.resolveSuggestiontoSkill(
+        'abc', 'pqr', 'accept', 'review message', 'easy', onSuccess, onFailure);
+      tick();
+
+      expect(carbas.reviewSkillSuggestionAsync)
+        .toHaveBeenCalledWith('abc', 'pqr', requestBody);
+      expect(onSuccess).not.toHaveBeenCalled();
+      expect(onFailure).toHaveBeenCalled();
+    }));
+  });
+
+  describe('updateTranslationSuggestionAsync', () => {
+    const requestBody = {
+      translation_html: '<p>In English</p>'
+    };
+
+    let onSuccess: jasmine.Spy<() => void>;
+    let onFailure: jasmine.Spy<(error: unknown) => void>;
+
+    beforeEach(() => {
+      onSuccess = jasmine.createSpy(
+        'onSuccess', () => {});
+      onFailure = jasmine.createSpy('onFailure', (error) => {});
+    });
+
+    it('should call onSuccess function when' +
+    'updateTranslationSuggestionAsync succeeds', fakeAsync(() => {
+      spyOn(carbas, 'updateTranslationSuggestionAsync').and
+        .returnValue(Promise.resolve());
+
+      cars.updateTranslationSuggestionAsync(
+        'pqr', '<p>In English</p>', onSuccess, onFailure);
+      tick();
+
+      expect(carbas.updateTranslationSuggestionAsync)
+        .toHaveBeenCalledWith('pqr', requestBody);
+      expect(onSuccess).toHaveBeenCalled();
+      expect(onFailure).not.toHaveBeenCalled();
+    }));
+
+    it('should call onFailure function when' +
+    'updateTranslationSuggestionAsync fails', fakeAsync(() => {
+      spyOn(carbas, 'updateTranslationSuggestionAsync').and
+        .returnValue(Promise.reject());
+
+      cars.updateTranslationSuggestionAsync(
+        'pqr', '<p>In English</p>', onSuccess, onFailure);
+      tick();
+
+      expect(carbas.updateTranslationSuggestionAsync)
+        .toHaveBeenCalledWith('pqr', requestBody);
+      expect(onSuccess).not.toHaveBeenCalled();
+      expect(onFailure).toHaveBeenCalled();
+    }));
+  });
+
+  describe('updateQuestionSuggestionAsync', () => {
+    const testQuestionStateData = {
```

Comment:
The "test" prefix seems redundant here. Can remove. Here and below.

Reply:
done